test/utils/audit.go

```diff
@@ -46,6 +52,52 @@ type AuditEvent struct {
 	RequestObject      bool
 	ResponseObject     bool
 	AuthorizeDecision  string
+
+	// The Check functions in this package takes ownerships of these maps. You should
+	// not reference these maps after calling the Check functions.
+	WebhookMutationAnnotations map[string]string
+	WebhookPatchAnnotations    map[string]string
+}
+
+// auditEvent is a private wrapper on top of AuditEvent used by auditEventTracker
+type auditEvent struct {
+	event AuditEvent
+	found bool
+}
+
+// auditEventTracker keeps track of AuditEvent expectations and marks matching events as found
+type auditEventTracker struct {
+	events []*auditEvent
+}
+
+// newAuditEventTracker creates a tracker that tracks whether expect events are found
+func newAuditEventTracker(expected []AuditEvent) *auditEventTracker {
+	expectations := &auditEventTracker{events: []*auditEvent{}}
+	for _, event := range expected {
+		// we copy the references to the maps in event
+		expectations.events = append(expectations.events, &auditEvent{event: event, found: false})
+	}
+	return expectations
+}
+
+// Match marks the given event as found if it's expected
+func (t *auditEventTracker) Match(event AuditEvent) {
```

Comment:
this does not match, it marks. The verb should be the first word in a go func with side effect.